services/horizon/internal/actions_trade.go

```
@@ -57,6 +57,25 @@ func (action *TradeIndexAction) loadRecords() {
 		trades.ForAccount(action.AccountFilter)
 	}
 
+	// If only one of the asset pair is provided, invalid parameter for request
+	if (action.HasBaseAssetFilter != action.HasCounterAssetFilter) {
+		if (!action.HasBaseAssetFilter) {
+			action.SetInvalidField(
+			"base_asset_type", errors.New("invalid asset type: was not one of 'native', 'credit_alphanum4', 'credit_alphanum12'"),
+			);
+		}
+
+		if (!action.HasCounterAssetFilter) {
+			action.SetInvalidField(
+				"counter_asset_type", errors.New("invalid asset type: was not one of 'native', 'credit_alphanum4', 'credit_alphanum12'"),
```

It appears that you will set this error even if `counter_asset_type` is present and legal.

For example, if the request has legal `base_asset_type`, `base_asset_issuer`,` base_asset_code` and `counter_asset_type`. However, it's missing `counter_asset_issuer` and `counter_asset_code`. In that case, you will rightfully return a 400 but with a wrong error message.

---

In that case, the request should have already failed. The asset filter is checked here: 
https://github.com/stellar/go/blob/7d8a7b7b744f11acc005f1c4199460bb74868eb5/services/horizon/internal/actions/helpers.go#L322

If asset_type is provided while other necessary fields are not, this function should throw a missing resource exception:

`func (base *Base) MaybeGetAsset(prefix string) (xdr.Asset, bool) {
}
`